### DISCUSSION TITLE
Benchmarking: comparing performance of different C# options and XAML

### DISCUSSION THREAD

**Discussion by mrlacey** [CommunityToolkit/Maui.Markup#204] [General]
I'm interested in comparing the performance differences among the different ways of setting up UI elements in C# and XAML.

For example, how would the performance of creating the following (admittedly very simple) examples compare?

```cs
new Button { Content = "click me" };
```

```cs
new Button.Content("click me");
```

```xml
<Button Content="click me" />
```

I'm interested in things more complicated but this is for the sake of a simple example.

I wondered if anyone here has done any micro-benchmarking to compare such things and can provide recommendations/advice/guidance/etc.

**Comment by bijington** [comment]
I would be keen to see the results of this discussion. I wonder if we could combine the device runners and benchmark .NET packages to truly get a sense of performance on the various platforms
